backend/src/whatsapp/middlewares/whatsapp.middleware.ts

```diff
@@ -1,24 +1,37 @@
 import { loggerWithLabel } from '@core/logger'
 import { Handler, Request, Response } from 'express'
-import { CredentialService } from '@core/services'
+import { CredentialService, DomainService } from '@core/services'
 import { WhatsappService } from '@whatsapp/services'
 
 export interface WhatsappMiddleware {
-  getCredentialsFromBody: Handler
+  getCredentials: Handler
   getCampaignDetails: Handler
-
+  getPhoneNumbers: Handler
+  getTemplates: Handler
   sendMessage: Handler
 }
 
 export const InitWhatsappMiddleware = (
   credentialService: CredentialService
 ): WhatsappMiddleware => {
   const logger = loggerWithLabel(module)
-  const getCredentialsFromBody = async (
+  const getCredentials = async (
     req: Request,
     res: Response
   ): Promise<void | Response> => {
-    logger.info(req, res, credentialService)
+    const userId = req.session?.user?.id
+    const userDomain = await DomainService.getUserDomain(userId)
+    // if empty or user not found
+    if (!userDomain) {
+      logger.error({
+        message: 'User does not belong to a domain',
+        action: 'getTemplates',
+      })
+      res.sendStatus(401)
```

Comment:
Let's have a response body that follows our standard fields of `code` and `message` fields. Come back to this after the refactor is merged in for error responses